### PR TITLE
Aj/covariant3vec

### PR DESCRIPTION
**Pull request by trontrytel**
## Purpose 
The goal is to change the vertical velocity type in CATs (Clima Atmos Turbulence Convection :) to be `Covariance3Vector` instead of a scalar. 

This is linked to SDI: https://github.com/CliMA/ClimaAtmos.jl/issues/789

## To-do
- [x] Make the code compile
- [ ] Make the simulation results unchanged

### CODE REVIEW

**Comment by trontrytel** [comment]
The code compiles, but the results are changed. Will try to solve it tomorrow

File: src/TurbulenceConvection/update_aux.jl [hunk 1]
```diff
@@ -152,15 +152,14 @@ function update_aux!(
...
-        wcomponent(CCG.WVector(prog_gm_f.w)) / (1 - Ifb(aux_bulk.area))
+    @. aux_en_f.w = prog_gm_f.w / (1 - Ifb(aux_bulk.area))
     @inbounds for i in 1:N_up
         @. aux_en_f.w -=
             Ifb(aux_up[i].area) * prog_up_f[i].w / (1 - Ifb(aux_bulk.area))
     end
 
     @. aux_en.e_kin =
-        LA.norm_sqr(C123(prog_gm_uₕ) + C123(Ic(wvec(aux_en_f.w)))) / 2
+        LA.norm_sqr(C123(prog_gm_uₕ) + C123(Ic(aux_en_f.w))) / 2
```
Comment (simonbyrne) [inline src/TurbulenceConvection/update_aux.jl:162]:
```suggestion
        LA.norm_sqr(C123(prog_gm_uₕ) + C123(Ic(CCG.WVector(aux_en_f.w)))) / 2
```

File: src/TurbulenceConvection/EDMF_functions.jl [hunk 2]
```diff
@@ -1153,15 +1166,15 @@ function compute_en_tendencies!(
...
     end
 
     RB = CCO.RightBiasedC2F(; top = CCO.SetValue(FT(0)))
     @. tend_covar =
         press + buoy + shear + entr_gain + rain_src - D_env * covar -
         (c_d * sqrt(max(tke_en, 0)) / max(aux_tc.mixing_length, 1)) *
-        prog_covar - ∇c(wvec(RB(prog_covar * Ic(w_en_f)))) +
-        ∇c(ρ_f * If(aeK) * ∇f(covar))
+        prog_covar - div_c(wvec(RB(prog_covar * Ic(w_en_f)))) +
+        div_c(ρ_f * If(aeK) * grad_f(covar))
```
Comment (simonbyrne) [inline src/TurbulenceConvection/EDMF_functions.jl:1177]:
```suggestion
        prog_covar - div_c(RB(prog_covar * Ic(CCG.WVector(w_en_f)))) +
        div_c(ρ_f * If(aeK) * grad_f(covar))
```

File: src/TurbulenceConvection/EDMF_functions.jl [hunk 3]
```diff
@@ -1153,15 +1166,15 @@ function compute_en_tendencies!(
         # TODO: using `Int(bool) *` means that NaNs can propagate
         # into the solution. Could we somehow call `ifelse` instead?
         @. D_env +=
-            Int(a_up > min_area) * ρ_c * a_up * Ic(w_up) * (entr_sc + turb_entr)
+            Int(a_up > min_area) * ρ_c * a_up * Ic(wcomponent(CCG.WVector(w_up))) * (entr_sc + turb_entr)
     end
 
     RB = CCO.RightBiasedC2F(; top = CCO.SetValue(FT(0)))
```
Comment (simonbyrne) [inline src/TurbulenceConvection/EDMF_functions.jl:1172]:
```suggestion
    RB = CCO.RightBiasedC2F(; top = CCO.SetValue(CCG.WVector(FT(0))))
```

**Comment by trontrytel** [comment]
Closing in favour of https://github.com/CliMA/ClimaAtmos.jl/pull/992

**Pull request closed without merging**